api/api.proto

```diff
@@ -2684,3 +2698,53 @@ message LocateResponse {
     // <containerid>: /var/www
     map<string, string> dockerids = 2;
 }
+
+message VolumePlacementSpec {
+  // Defines a list of rules as part of the placement spec
+  repeated VolumePlacementRule rules = 1;
+}
+
+message VolumePlacementRule {
+  // Defines the number of volume replicas affected by this rule. If not provided,
+  // rule would affect all replicas
+  // (optional)
+  int64 affected_replicas = 1;
+  // Defines the weight of the rule which allows to break to tie with other matching rules
+  // (optional)
+  int64 weight = 2;
+  // Specifies the rule enforcement policy. Can take values: required or preferred.
+  // (optional)
+  VolumePlacementEnforcementType enforcement = 3;
+  // Defines the affinity rule for the volume placement
+  AffinityRule affinity = 4;
+  // Defines the anti affinity rule for the volume placement
+  AffinityRule anti_affinity = 5;
+}
+
+message AffinityRule {
+  // This is a list of label selector requirements. The requirements are ANDed.
+  repeated LabelSelectorRequirement match_expressions = 1;
+}
+
+
+// A label selector is a label query over a set of resources. The result of match_labels and
+// match_expressions are ANDed. A nil or empty label selector matches all objects.
+message LabelSelector {
```

Comment:
Should this a `LabelSelectorAny` ? Or is it missing value? or is this supposed to be an enum?

Reply:
This was a remnant of the previous change. `LabelSelector` is no longer needed. Removed it.